Facade_Engine/Compute/AdjacentElements.cs

```diff
@@ -87,6 +88,30 @@ public static List<IElement2D> AdjacentElements(this IElement1D elem, List<IElem
 
             return adjElems;
         }
+
+
+        /***************************************************/
+
+        [Description("Returns elements adjacent to a provided Element1D from a collection of elements.")]
+        [Input("elem", "Element to find adjacencies at.")]
+        [Input("refElems", "Elements to use to find element adjacencies.")]
+        [Output("adjElems", "Adjacent elements to the provided element.")]
+        public static List<IElement1D> AdjacentElements(this IElement1D elem, List<IElement1D> refElems)
+
+        {
+            List<IElement1D> adjElems = new List<IElement1D>();
+
+            foreach (IElement1D refElem in refElems)
+            {
+                if (Query.IIsAdjacent(refElem, elem))
+                {
+                    adjElems.Add(refElem);
+                }
+            }
+
+            // Return the adjacencies
+            return adjElems;
```

Comment:
How about:
```suggestion
            return refElems.Where(x => Query.IIsAdjacent(x, elem)).ToList();
```

If `IIsAdjacent` is an extension method (it should be) then you could do:

```suggestion
            return refElems.Where(x => x.IIsAdjacent(elem)).ToList();
```